vulnerabilities/models.py

```diff
@@ -2744,6 +2748,67 @@ def url(self):
             return f"https://github.com/nodejs/security-wg/blob/main/vuln/npm/{id}.json"
 
 
+class Patch(models.Model):
+    """
+    A generic patch linked to an advisory.
+    Used for raw text patches, fallback scenarios, or unsupported VCS types.
+    """
+
+    patch_url = models.URLField(
+        null=True,
+        blank=True,
+        help_text="URL to the patch file or diff.",
+    )
+
+    patch_text = models.TextField(
+        null=True,
+        blank=True,
+        help_text="The actual text content of the code patch/diff.",
+    )
+
+    patch_checksum = models.CharField(
+        max_length=128,
+        blank=True,
+        null=True,
+        help_text="SHA512 checksum of the patch content.",
+    )
+
+    def save(self, *args, **kwargs):
+        if self.patch_text:
+            self.patch_checksum = compute_patch_checksum(self.patch_text)
+        super().save(*args, **kwargs)
+
+    class Meta:
+        unique_together = ["patch_checksum", "patch_url"]
```

Comment:
Since both `patch_url` and `patch_text` could be blank we should put some constraint here to make sure we have at least one field to create a valid Patch obj.